Source: Rasmus Thomsen <oss@cogitri.dev>
Upstream: No, Alpine/Musl specific hack
Reason: rofiles-fuse is always broken on musl as of now, so
disable it for know so users don't get (very weird) failures
when building flatpaks if they forget to specify that opt.
See https://github.com/flatpak/flatpak-builder/issues/329
for more info.
diff --git a/src/builder-main.c b/src/builder-main.c
index a5f6601..f63c42f 100644
--- a/src/builder-main.c
+++ b/src/builder-main.c
@@ -98,7 +98,7 @@ static GOptionEntry entries[] = {
   { "ccache", 0, 0, G_OPTION_ARG_NONE, &opt_ccache, "Use ccache", NULL },
   { "disable-cache", 0, 0, G_OPTION_ARG_NONE, &opt_disable_cache, "Disable cache lookups", NULL },
   { "disable-tests", 0, 0, G_OPTION_ARG_NONE, &opt_disable_tests, "Don't run tests", NULL },
-  { "disable-rofiles-fuse", 0, 0, G_OPTION_ARG_NONE, &opt_disable_rofiles, "Disable rofiles-fuse use", NULL },
+  { "disable-rofiles-fuse", 0, 0, G_OPTION_ARG_NONE, &opt_disable_rofiles, "Disable rofiles-fuse use (Note: This is ALWAYS disabled on VoidLinux with Musl due to https://github.com/flatpak/flatpak-builder/issues/329)", NULL },
   { "disable-download", 0, 0, G_OPTION_ARG_NONE, &opt_disable_download, "Don't download any new sources", NULL },
   { "disable-updates", 0, 0, G_OPTION_ARG_NONE, &opt_disable_updates, "Only download missing sources, never update to latest vcs version", NULL },
   { "download-only", 0, 0, G_OPTION_ARG_NONE, &opt_download_only, "Only download sources, don't build", NULL },
@@ -508,7 +508,7 @@ main (int    argc,
 
   build_context = builder_context_new (cwd_dir, app_dir, opt_state_dir);
 
-  builder_context_set_use_rofiles (build_context, !opt_disable_rofiles);
+  builder_context_set_use_rofiles (build_context, 0);
   builder_context_set_run_tests (build_context, !opt_disable_tests);
   builder_context_set_no_shallow_clone (build_context, opt_no_shallow_clone);
   builder_context_set_keep_build_dirs (build_context, opt_keep_build_dirs);
